translate/cloud-client/src/main/java/com/example/translate/HybridGlossaries.java

```diff
@@ -0,0 +1,301 @@
+import com.google.cloud.vision.v1.AnnotateImageResponse;
+import com.google.cloud.vision.v1.BatchAnnotateImagesResponse;
+import com.google.cloud.vision.v1.Feature;
+import com.google.cloud.vision.v1.Feature.Type;
+import com.google.cloud.vision.v1.Image;
+import com.google.cloud.vision.v1.ImageAnnotatorClient;
+
+import com.google.common.html.HtmlEscapers;
+import com.google.protobuf.ByteString;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+// [END translate_hybrid_imports]
+
+
+public class HybridGlossaries {
+
+  // [START translate_hybrid_project_id]
+  private static final String projectId = System.getenv("PROJECT_ID");
+  // [END translate_hybrid_project_id]
+
+  // [START translate_hybrid_vision]
+  /**
+   * @param filePath input image file
+   **/
+  static String picToText(String filePath) {
+    List<AnnotateImageRequest> requests = new ArrayList<>();
+
+    try (ImageAnnotatorClient client = ImageAnnotatorClient.create()) {
+      ByteString imageBytes = ByteString.readFrom(new FileInputStream(filePath));
+
+      Image image = Image.newBuilder().setContent(imageBytes).build();
+
+      Feature feature = Feature.newBuilder().setType(Type.DOCUMENT_TEXT_DETECTION).build();
+
+      AnnotateImageRequest request =
+              AnnotateImageRequest.newBuilder()
+                      .addFeatures(feature)
+                      .setImage(image)
+                      .build();
+
+      requests.add(request);
+
+      BatchAnnotateImagesResponse response = client.batchAnnotateImages(requests);
+      List<AnnotateImageResponse> responses = response.getResponsesList();
+
+      StringBuilder result = new StringBuilder();
+
+      for (AnnotateImageResponse res : responses) {
+        result.append(String.format("%s\n", res.getFullTextAnnotation().getText()));
+      }
+
+      return result.toString();
+    } catch (IOException e) {
+      return String.format("Failed to get the text from the image.\n%s\n", e.getMessage());
+    }
+  }
+  // [END translate_hybrid_vision]
+
+  // [START translate_hybrid_create_glossary]
+  /** Creates a GCP glossary resource
+   * Assumes you've already manually uploaded a glossary to Cloud Storage
+   *
+   * ARGS
+   * @param languages list of languages in the glossary
+   * @param projectId GCP project id
+   * @param glossaryDisplayName name you want to give this glossary resource
+   * @param glossaryUri the uri of the glossary you uploaded to Cloud Storage
+   * 
+   **/
+  static void createGlossary(List<String> languages, String projectId,
+                             String glossaryDisplayName, String glossaryUri) {
+    try (TranslationServiceClient translationServiceClient = TranslationServiceClient.create()) {
+
+      LocationName locationName =
+              LocationName.newBuilder().setProject(projectId).setLocation("us-central1").build();
```

Comment:
I don't know if there is still the possibility to edit this library's API, but having builders for these type of variables is an anti-pattern. 

Builders are useful when you have a bunch of optional arguments and don't want to have a bunch of overloaded APIs. These arguments are all required, so it should just have a method name. 

The API for this should be a static method: `LocationName.of(projectId, "us-central-1");` (this is way nicer to write and way easier to read).

Reply:
Updated